packages/design-system/src/components/web-components/ds-alert.test.tsx

```diff
@@ -0,0 +1,180 @@
+import React from 'react';
+import Alert, { AlertProps } from '../Alert/Alert';
+import { UtagContainer } from '../analytics';
+import { setAlertSendsAnalytics } from '../flags';
+import { render, screen } from '@testing-library/react';
+import register from 'preact-custom-element';
+
+register(Alert, 'ds-alert');
+
+// Typically, defining this interface would allow for kebab-case props, but isn't on Alert for some reason.
+declare global {
+  namespace JSX {
+    interface IntrinsicElements {
+      'ds-alert': AlertProps;
+    }
+  }
+}
+
+const defaultText = 'Ruhroh';
+
+function renderAlert(props: AlertProps = {}) {
+  return render(<ds-alert {...props}>{defaultText}</ds-alert>);
+}
+
+function expectHasClass(className: string) {
+  expect(screen.getByRole('region').className).toContain(className);
+}
+
+describe('Alert', function () {
+  // WC renders <slot> in snaps
+  it('renders alert', () => {
+    renderAlert({ id: 'static-id' });
+    const alert = screen.getByRole('region');
+    expect(alert).toMatchSnapshot();
```

Comment:
i figured out why i didn't notice the root web component elements in our original snap tests - we're querying child elements within the root component before taking the snap. as a result, we only get the guts of a web component captured in the snaps. it's the same with our react components. i wanted to maintain some parity between the two tests, so i kept the results as is, but this does seem like something to look into for both web component and react tests imo.